libs/evcs-service/src/main/java/uk/gov/di/ipv/core/library/service/EvcsService.java

```diff
@@ -139,26 +148,51 @@ private void persistEvcsUserVCs(
                             .toList();
-            var existingPendingReturnEvcsUserVcsToUpdate =
+            var existingPendingReturnUserVcsNotInSessionToUpdate =
                     existingEvcsUserVCs.stream()
                             .filter(vc -> vc.state().equals(PENDING_RETURN))
+                            .filter(
+                                    vc ->
+                                            credentials.stream()
+                                                    .noneMatch(
+                                                            credential ->
+                                                                    credential
+                                                                            .getVcString()
+                                                                            .equals(vc.vc())))
                             .map(
                                     vc ->
                                             new EvcsUpdateUserVCsDto(
                                                     getVcSignature(vc),
                                                     EvcsVCState.ABANDONED,
                                                     null))
                             .toList();
-            var evcsUserVCsToUpdate =
+            var existingPendingReturnUserVcsInSessionToUpdate =
+                    existingEvcsUserVCs.stream()
+                            .filter(vc -> vc.state().equals(PENDING_RETURN))
+                            .filter(
+                                    vc ->
+                                            credentials.stream()
+                                                    .anyMatch(
+                                                            credential ->
+                                                                    credential
+                                                                            .getVcString()
+                                                                            .equals(vc.vc())))
+                            .map(
+                                    vc ->
+                                            new EvcsUpdateUserVCsDto(
+                                                    getVcSignature(vc), EvcsVCState.CURRENT, null))
+                            .toList();
+            var userVCsToUpdate =
                     Stream.concat(
-                                    existingCurrentEvcsUserVcsToUpdate.stream(),
-                                    existingPendingReturnEvcsUserVcsToUpdate.stream())
+                                    Stream.concat(
+                                            existingCurrentUserVcsNotInSessionToUpdate.stream(),
+                                            existingPendingReturnUserVcsNotInSessionToUpdate
+                                                    .stream()),
+                                    existingPendingReturnUserVcsInSessionToUpdate.stream())
```

Comment:
(If you drop the `.toList()`s on the variables, then you don't need the `.stream()`s here)